virtual_ship/make_realistic/ctd_make_realistic.py

```diff
@@ -0,0 +1,187 @@
+            times=reltimesec,
+            temperatures=temperature,
+            depths=depth,
+            salinities=salinity,
+        )
+
+        with open(out_file, "w") as out_cnv:
+            out_cnv.write(cnv)
+
+    return files
+
+
+def _add_temperature_noise(temperature: np.ndarray, depth: np.ndarray) -> np.ndarray:
+    surface_noise = (
+        2.0
+        * (np.random.random_sample(len(temperature)) * 2.0 - 1)
+        * np.exp(depth / 4.0)
+    )
+    noise = (
+        0.05
+        * np.maximum(1.0 + depth / 1000.0, 0.1)
+        * np.random.random_sample(len(temperature))
+    )
+    drift = (
+        0.3
+        * (np.random.random() * 2.0 - 1)
+        * np.array(range(len(temperature)))
+        / len(temperature)
+    )
+    opensimplex.seed(np.random.randint(999999))
+    noise2 = (
+        2.0
+        * np.maximum(1.0 + depth / 1000.0, 0.1)
+        * (
+            np.array([opensimplex.noise2(n / 30, 0.0) for n in range(len(temperature))])
+            ** 8
+        )
+    )
+    opensimplex.seed(np.random.randint(999999))
+    noise3 = 0.1 * (
+        np.array([opensimplex.noise2(n / 8, 0.0) for n in range(len(temperature))]) ** 2
+    )
+
+    return temperature + surface_noise + drift + noise + noise3 + noise2
+
+
+def _add_salinity_noise(salinity: np.ndarray, depth: np.ndarray) -> np.ndarray:
+    surface_noise = (
+        2.0 * (-np.random.random_sample(len(salinity))) * np.exp(depth / 4.0)
+    )
+    noise = (
+        0.001
+        * np.maximum(1.0 + depth / 1000.0, 0.1)
+        * np.random.random_sample(len(salinity))
+    )
+    drift = (
+        0.02
+        * (np.random.random() * 2.0 - 1)
+        * np.array(range(len(salinity)))
+        / len(salinity)
+    )
+    opensimplex.seed(np.random.randint(999999))
+    noise2 = (
+        0.07
+        * np.maximum(1.0 + depth / 1000.0, 0.1)
+        * (
+            np.array([opensimplex.noise2(n / 30, 0.0) for n in range(len(salinity))])
+            ** 8
+        )
+    )
+    opensimplex.seed(np.random.randint(999999))
+    noise3 = 0.003 * (
+        np.array([opensimplex.noise2(n / 8, 0.0) for n in range(len(salinity))]) ** 2
+    )
+
+    return salinity + surface_noise + drift + noise + noise2 + noise3
+
+
+def _to_cnv(
+    filename: str,
+    latitudes: np.ndarray,
+    longitudes: np.ndarray,
+    times: np.ndarray,
+    temperatures: np.ndarray,
+    depths: np.ndarray,
+    salinities: np.ndarray,
+) -> str:
+    """
+    Convert data to CNV.
+
+    :param filename: Output file name.
+    :param latitudes: Latitude data.
+    :param longitudes: Longitude data.
+    :param times: Elapsed time since first measurement in seconds.
+    :param temperatures: Temperature data.
+    :param depths: Depth data.
+    :param salinities: Salinity data.
+    :returns: The CNV.
+
+    """
+    header = rf"""
+* Sea-Bird SBE 9 Data File:
+* FileName = {filename}
+# start_time = Aug 02 2024 21:29:45 [NMEA time, first data scan]
```

Comment:
Should this starttime not be taken from the virtual ship cast time?

Reply:
Done